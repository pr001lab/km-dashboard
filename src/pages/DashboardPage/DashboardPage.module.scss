@import '../../styles/_vars.scss';

.page-info {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 150px;

  & .page-info__text {
    color: $text-heading;
    text-align: center;
    font-family: $headingFontFamily;
    font-size: 18px;
    font-weight: 500;
    line-height: normal;
    margin-bottom: 42px;
  }
}

.wrapper-input {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 45px;
  border-radius: 15px;
  border: 1px solid #ebebeb;
  background: $white;
  margin-top: 40px;
  margin-bottom: 20px;

  & .label-input {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 45px;
    margin-left: 13px;
    margin-right: 11px;
  }

  & .input-info {
    color: $gray;
    font-family: $defaultFontFamily;
    font-size: 14px;
    font-weight: 400;
    line-height: normal;
    text-wrap: nowrap;
    margin-left: 11px;
    margin-right: 17px;
  }

  &:focus-within,
  &:active {
    outline: 2px solid #8686ff;
  }
}

input {
  font-family: $defaultFontFamily;
  color: $text-primary;
  font-size: 14px;
  font-weight: 400;
  line-height: normal;
  width: 100%;
  border: none;
  background: transparent;
  outline: none;

  &::placeholder {
    color: $gray;
    font-family: $defaultFontFamily;
    font-size: 14px;
    font-weight: 400;
    line-height: normal;
  }
}

.table {
  font-family: $defaultFontFamily;
  display: flex;
  flex-direction: column;
}
