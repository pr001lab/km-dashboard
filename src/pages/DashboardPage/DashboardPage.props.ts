import { ComponentProps } from 'react';

export interface DashboardPageProps extends ComponentProps<'div'> {}
