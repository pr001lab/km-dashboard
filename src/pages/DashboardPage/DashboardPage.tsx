import { useState } from 'react';
import Heading from '../../components/common/Heading/Heading';
import { fieldSort, orderSortStatus, SortEnum } from '../../constant';
import styles from '../DashboardPage/DashboardPage.module.scss';
import Wrapper from '../../components/common/Wrapper/Wrapper';
import { ReactComponent as Search } from '../../assets/icons/Search.svg';
import Loader from '../../components/common/Loader/Loader';
import Button from '../../components/common/Button/Button';
import { useGetData } from '../../hooks/useGetData';
import TBody from '../../components/TBody/TBody';
import THead from '../../components/THead/THead';
import { Test } from '../../types';

function DashboardPage() {
  const { tests, loading } = useGetData();
  const [search, setSearch] = useState('');
  const [sortOrder, setSortOrder] = useState<SortEnum>(SortEnum.None);
  const [sortName, setSortName] = useState<string>('name');

  const testFilter = (test: Test) => {
    if (search === '') {
      return true;
    }
    return test.name.toLowerCase().includes(search.toLowerCase());
  };

  const sortData = (a: { [key: string]: any }, b: { [key: string]: any }) => {
    if (sortName === fieldSort.STATUS) {
      if (sortOrder === SortEnum.Ascending) {
        return (
          orderSortStatus.indexOf(a.status) - orderSortStatus.indexOf(b.status)
        );
      } else if (sortOrder === SortEnum.Descending) {
        return (
          orderSortStatus.slice().reverse().indexOf(a.status) -
          orderSortStatus.slice().reverse().indexOf(b.status)
        );
      }
      return 0;
    } else {
      if (sortOrder === SortEnum.Ascending) {
        return a[sortName].localeCompare(b[sortName]);
      } else if (sortOrder === SortEnum.Descending) {
        return b[sortName].localeCompare(a[sortName]);
      }
      return 0;
    }
  };

  const setColSort = (sortColName: string) => {
    if (sortName === sortColName) {
      if (sortOrder === SortEnum.None) {
        setSortOrder(SortEnum.Ascending);
      } else if (sortOrder === SortEnum.Ascending) {
        setSortOrder(SortEnum.Descending);
      } else {
        setSortOrder(SortEnum.None);
      }
    } else {
      setSortName(sortColName);
      setSortOrder(SortEnum.Ascending);
    }
  };

  const testResults = tests.filter(testFilter).slice().sort(sortData);

  if (loading) {
    return <Loader />;
  }

  return (
    <>
      <Heading level="h1">Dashboard</Heading>
      <Wrapper className={styles['wrapper-input']}>
        <label className={styles['label-input']} htmlFor="input">
          <Search />
        </label>
        <input
          placeholder="What test are you looking for?"
          id="input"
          value={search}
          onChange={(e) => setSearch(e.target.value)}
        />
        <Wrapper className={styles['input-info']}>
          {`${tests.filter(testFilter).length} tests`}
        </Wrapper>
      </Wrapper>
      {tests.filter(testFilter).length === 0 && (
        <Wrapper className={styles['page-info']}>
          <Wrapper className={styles['page-info__text']}>
            Your search did not match any results.
          </Wrapper>
          <Button onClick={() => setSearch('')}>Reset</Button>
        </Wrapper>
      )}
      {tests.filter(testFilter).length > 0 && (
        <table className={styles['table']}>
          <THead
            sortName={sortName}
            sortOrder={sortOrder}
            setColSort={setColSort}
          />
          <TBody tests={testResults} />
        </table>
      )}
    </>
  );
}

export default DashboardPage;
