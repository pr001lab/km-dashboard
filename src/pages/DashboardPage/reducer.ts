import { SortEnum } from '../../constant';
import { INITIAL_STATE } from './DashboardPage';

export const reducer = (
  state: typeof INITIAL_STATE,
  action: Record<string, string>,
) => {
  if (state.colName === action.colName) {
    if (state.sortOrder === SortEnum.None) {
      return {
        ...state,
        sortOrder: SortEnum.Ascending,
      };
    } else if (state.sortOrder === SortEnum.Ascending) {
      return {
        ...state,
        sortOrder: SortEnum.Descending,
      };
    } else {
      return {
        ...state,
        sortOrder: SortEnum.None,
      };
    }
  } else {
    return {
      ...state,
      colName: action.colName,
      sortOrder: SortEnum.Ascending,
    };
  }
};
