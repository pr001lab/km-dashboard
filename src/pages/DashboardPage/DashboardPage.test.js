import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import DashboardPage from './DashboardPage';
import { mockTests } from '../../../mocks/mocks';

const mockDataTests = {
  tests: mockTests,
  loading: false,
};

jest.mock('../../hooks/useGetData', () => ({
  useGetData: () => mockDataTests,
}));

describe('Router tests', () => {
  const renderWithRouter = () => {
    return {
      ...render(
        <MemoryRouter
          future={{
            v7_relativeSplatPath: true,
            v7_startTransition: true,
          }}
        >
          <DashboardPage />
        </MemoryRouter>,
      ),
    };
  };

  it('should render Loader for DashboardPage', () => {
    mockDataTests.loading = true;
    renderWithRouter();
    const divs = screen.getAllByRole('generic');
    expect(divs[1].classList.contains('loading-container')).toBe(true);
  });

  it('should render DashboardPage for the main path with data table', () => {
    mockDataTests.loading = false;
    renderWithRouter();
    const rows = screen.getAllByRole('row');

    expect(rows).toHaveLength(mockTests.length + 1);
  });

  it('should render DashboardPage for the main path without data table', () => {
    mockDataTests.tests = [];
    renderWithRouter();
    expect(
      screen.getByText(/Your search did not match any results./i),
    ).toBeInTheDocument();
  });
});
