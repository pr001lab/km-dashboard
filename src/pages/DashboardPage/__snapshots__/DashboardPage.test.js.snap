// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router tests should render DashboardPage for the main path with data rowsTable 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="heading"
      >
        Dashboard
      </h1>
      <div
        class="wrapper wrapper-input"
      >
        <label
          class="label-input"
          for="input"
        >
          <svg>
            Search.svg
          </svg>
        </label>
        <input
          id="input"
          placeholder="What test are you looking for?"
          value=""
        />
        <div
          class="wrapper input-info"
        >
          7 tests
        </div>
      </div>
      <table
        class="table"
      >
        <thead>
          <tr>
            <th />
            <th>
              <button
                class="button button table-header None"
              >
                Name
                <svg>
                  Chevron-sm.svg
                </svg>
              </button>
            </th>
            <th>
              <button
                class="button button table-header None"
              >
                Type
                <svg>
                  Chevron-sm.svg
                </svg>
              </button>
            </th>
            <th>
              <button
                class="button button table-header None"
              >
                Status
                <svg>
                  Chevron-sm.svg
                </svg>
              </button>
            </th>
            <th>
              <button
                class="button button table-header None"
              >
                Site
                <svg>
                  Chevron-sm.svg
                </svg>
              </button>
            </th>
            <th />
          </tr>
        </thead>
        <tbody>
          <tr
            class="row-table"
          >
            <td
              class="row-indicator siteId-2"
            />
            <td
              class="row-name"
            >
              Prototype of the new map
            </td>
            <td
              class="row-type"
            >
              Classic
            </td>
            <td
              class="row-status PAUSED"
            >
              Paused
            </td>
            <td
              class="row-site"
            />
            <td>
              <a
                class="link-button"
                href="/results/1"
              >
                Results
              </a>
            </td>
          </tr>
          <tr
            class="row-table"
          >
            <td
              class="row-indicator siteId-3"
            />
            <td
              class="row-name"
            >
              Dark theme test
            </td>
            <td
              class="row-type"
            >
              MVT
            </td>
            <td
              class="row-status DRAFT"
            >
              Draft
            </td>
            <td
              class="row-site"
            />
            <td>
              <a
                class="link-button gray-dark"
                href="/finalize/2"
              >
                Finalize
              </a>
            </td>
          </tr>
          <tr
            class="row-table"
          >
            <td
              class="row-indicator siteId-1"
            />
            <td
              class="row-name"
            >
              New Year's Sale
            </td>
            <td
              class="row-type"
            >
              MVT
            </td>
            <td
              class="row-status STOPPED"
            >
              Stopped
            </td>
            <td
              class="row-site"
            />
            <td>
              <a
                class="link-button"
                href="/results/3"
              >
                Results
              </a>
            </td>
          </tr>
          <tr
            class="row-table"
          >
            <td
              class="row-indicator siteId-1"
            />
            <td
              class="row-name"
            >
              Order basket redesing
            </td>
            <td
              class="row-type"
            >
              Classic
            </td>
            <td
              class="row-status ONLINE"
            >
              Online
            </td>
            <td
              class="row-site"
            />
            <td>
              <a
                class="link-button"
                href="/results/4"
              >
                Results
              </a>
            </td>
          </tr>
          <tr
            class="row-table"
          >
            <td
              class="row-indicator siteId-2"
            />
            <td
              class="row-name"
            >
              Spring promotion
            </td>
            <td
              class="row-type"
            >
              Server-side
            </td>
            <td
              class="row-status DRAFT"
            >
              Draft
            </td>
            <td
              class="row-site"
            />
            <td>
              <a
                class="link-button gray-dark"
                href="/finalize/5"
              >
                Finalize
              </a>
            </td>
          </tr>
          <tr
            class="row-table"
          >
            <td
              class="row-indicator siteId-3"
            />
            <td
              class="row-name"
            >
              Prototype of a new header
            </td>
            <td
              class="row-type"
            >
              Server-side
            </td>
            <td
              class="row-status ONLINE"
            >
              Online
            </td>
            <td
              class="row-site"
            />
            <td>
              <a
                class="link-button"
                href="/results/6"
              >
                Results
              </a>
            </td>
          </tr>
          <tr
            class="row-table"
          >
            <td
              class="row-indicator siteId-1"
            />
            <td
              class="row-name"
            >
              New Year's Sale Copy 1
            </td>
            <td
              class="row-type"
            >
              MVT
            </td>
            <td
              class="row-status DRAFT"
            >
              Draft
            </td>
            <td
              class="row-site"
            />
            <td>
              <a
                class="link-button gray-dark"
                href="/finalize/7"
              >
                Finalize
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <h1
      class="heading"
    >
      Dashboard
    </h1>
    <div
      class="wrapper wrapper-input"
    >
      <label
        class="label-input"
        for="input"
      >
        <svg>
          Search.svg
        </svg>
      </label>
      <input
        id="input"
        placeholder="What test are you looking for?"
        value=""
      />
      <div
        class="wrapper input-info"
      >
        7 tests
      </div>
    </div>
    <table
      class="table"
    >
      <thead>
        <tr>
          <th />
          <th>
            <button
              class="button button table-header None"
            >
              Name
              <svg>
                Chevron-sm.svg
              </svg>
            </button>
          </th>
          <th>
            <button
              class="button button table-header None"
            >
              Type
              <svg>
                Chevron-sm.svg
              </svg>
            </button>
          </th>
          <th>
            <button
              class="button button table-header None"
            >
              Status
              <svg>
                Chevron-sm.svg
              </svg>
            </button>
          </th>
          <th>
            <button
              class="button button table-header None"
            >
              Site
              <svg>
                Chevron-sm.svg
              </svg>
            </button>
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr
          class="row-table"
        >
          <td
            class="row-indicator siteId-2"
          />
          <td
            class="row-name"
          >
            Prototype of the new map
          </td>
          <td
            class="row-type"
          >
            Classic
          </td>
          <td
            class="row-status PAUSED"
          >
            Paused
          </td>
          <td
            class="row-site"
          />
          <td>
            <a
              class="link-button"
              href="/results/1"
            >
              Results
            </a>
          </td>
        </tr>
        <tr
          class="row-table"
        >
          <td
            class="row-indicator siteId-3"
          />
          <td
            class="row-name"
          >
            Dark theme test
          </td>
          <td
            class="row-type"
          >
            MVT
          </td>
          <td
            class="row-status DRAFT"
          >
            Draft
          </td>
          <td
            class="row-site"
          />
          <td>
            <a
              class="link-button gray-dark"
              href="/finalize/2"
            >
              Finalize
            </a>
          </td>
        </tr>
        <tr
          class="row-table"
        >
          <td
            class="row-indicator siteId-1"
          />
          <td
            class="row-name"
          >
            New Year's Sale
          </td>
          <td
            class="row-type"
          >
            MVT
          </td>
          <td
            class="row-status STOPPED"
          >
            Stopped
          </td>
          <td
            class="row-site"
          />
          <td>
            <a
              class="link-button"
              href="/results/3"
            >
              Results
            </a>
          </td>
        </tr>
        <tr
          class="row-table"
        >
          <td
            class="row-indicator siteId-1"
          />
          <td
            class="row-name"
          >
            Order basket redesing
          </td>
          <td
            class="row-type"
          >
            Classic
          </td>
          <td
            class="row-status ONLINE"
          >
            Online
          </td>
          <td
            class="row-site"
          />
          <td>
            <a
              class="link-button"
              href="/results/4"
            >
              Results
            </a>
          </td>
        </tr>
        <tr
          class="row-table"
        >
          <td
            class="row-indicator siteId-2"
          />
          <td
            class="row-name"
          >
            Spring promotion
          </td>
          <td
            class="row-type"
          >
            Server-side
          </td>
          <td
            class="row-status DRAFT"
          >
            Draft
          </td>
          <td
            class="row-site"
          />
          <td>
            <a
              class="link-button gray-dark"
              href="/finalize/5"
            >
              Finalize
            </a>
          </td>
        </tr>
        <tr
          class="row-table"
        >
          <td
            class="row-indicator siteId-3"
          />
          <td
            class="row-name"
          >
            Prototype of a new header
          </td>
          <td
            class="row-type"
          >
            Server-side
          </td>
          <td
            class="row-status ONLINE"
          >
            Online
          </td>
          <td
            class="row-site"
          />
          <td>
            <a
              class="link-button"
              href="/results/6"
            >
              Results
            </a>
          </td>
        </tr>
        <tr
          class="row-table"
        >
          <td
            class="row-indicator siteId-1"
          />
          <td
            class="row-name"
          >
            New Year's Sale Copy 1
          </td>
          <td
            class="row-type"
          >
            MVT
          </td>
          <td
            class="row-status DRAFT"
          >
            Draft
          </td>
          <td
            class="row-site"
          />
          <td>
            <a
              class="link-button gray-dark"
              href="/finalize/7"
            >
              Finalize
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Router tests should render DashboardPage for the main path without data table 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="heading"
      >
        Dashboard
      </h1>
      <div
        class="wrapper wrapper-input"
      >
        <label
          class="label-input"
          for="input"
        >
          <svg>
            Search.svg
          </svg>
        </label>
        <input
          id="input"
          placeholder="What test are you looking for?"
          value=""
        />
        <div
          class="wrapper input-info"
        >
          0 tests
        </div>
      </div>
      <div
        class="wrapper page-info"
      >
        <div
          class="wrapper page-info__text"
        >
          Your search did not match any results.
        </div>
        <button
          class="button"
        >
          Reset
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <h1
      class="heading"
    >
      Dashboard
    </h1>
    <div
      class="wrapper wrapper-input"
    >
      <label
        class="label-input"
        for="input"
      >
        <svg>
          Search.svg
        </svg>
      </label>
      <input
        id="input"
        placeholder="What test are you looking for?"
        value=""
      />
      <div
        class="wrapper input-info"
      >
        0 tests
      </div>
    </div>
    <div
      class="wrapper page-info"
    >
      <div
        class="wrapper page-info__text"
      >
        Your search did not match any results.
      </div>
      <button
        class="button"
      >
        Reset
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Router tests should render Loader for DashboardPage 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="loading-container"
      >
        <div
          class="loading-spinner"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="loading-container"
    >
      <div
        class="loading-spinner"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
